contracts/rewards/StableLPMaker.sol

```diff
@@ -0,0 +1,172 @@
+
+    event LogLpTokensSentTopTRI(uint256 tlpAmount);
+    event LogLpTokensSentToDao(uint256 daoAmount);
+    event LogProtocolOwnedLiquidity(uint256 oldpolPercent, uint256 newStableSwap);
+
+    constructor(
+        address _stableSwap,
+        address _pTri,
+        address _usn,
+        address _usdt,
+        address _usdc ,
+        address _tlpToken,
+        address _dao
+    ) public {
+        stableSwap = ISwap(_stableSwap);
+        pTri = _pTri;
+        usn = _usn;
+        usdt = _usdt;
+        usdc = _usdc;
+        tlpToken = _tlpToken;
+        dao = _dao;
+    }
+
+    function setpTri(address _pTri) public onlyOwner {
+        address oldpTri;
+        oldpTri = pTri;
+        pTri = _pTri;
+
+        emit LogSetpTri(oldpTri, _pTri);
+    }
+
+    function setprotocolOwnerLiquidityPercent(uint256 _polPercent) public onlyOwner {
+        require(_polPercent < 50, "POL is too high");
+        uint256 oldpolPercent;
+        oldpolPercent = polPercent;
+        _polPercent = _polPercent;
+
+        emit LogProtocolOwnedLiquidity(oldpolPercent, _polPercent);
+    }
+
+
+    // C6: It's not a fool proof solution, but it prevents flash loans, so here it's ok to use tx.origin
+    modifier onlyEOA() {
+        // Try to make flash-loan exploit harder to do by only allowing externally owned addresses.
+        require(msg.sender == tx.origin, "StableLpMaker: must use EOA");
+        _;
+    }
+
+    function withdrawStableTokenFees() public onlyEOA {
+        // Withdraw admin fees from the Stableswap Pool to stable tokens
+        stableSwap.withdrawAdminFees();
+
+        emit LogWithdrawFees();
+    }
+
+    // This onyl works for USDC and USDT. If we have more tokens, need to make this more dynamic
+    function swapStableTokensToUsn() public onlyEOA {
+        address[2] memory stableTokensToSwapToUsn = [usdc, usdt];
+        for (uint256 i = 0; i < stableTokensToSwapToUsn.length; i++) {
+            IERC20(stableTokensToSwapToUsn[i]).approve(
+                address(stableSwap),
+                IERC20(stableTokensToSwapToUsn[i]).balanceOf(address(this))
+            );
+
+            uint256 stableTokenAmount = IERC20(stableTokensToSwapToUsn[i]).balanceOf(address(this));
+
+            stableSwap.swap(
+                stableSwap.getTokenIndex(stableTokensToSwapToUsn[i]),
+                stableSwap.getTokenIndex(usn),
+                stableTokenAmount,
+                0,
+                block.timestamp + 60
+            );
+
+            emit LogSwapStableTokenToUsn(stableTokensToSwapToUsn[i], stableTokenAmount);
+        }
+    }
+
+    function addLiquidityToStableSwap() public onlyEOA {
+        uint256 usnAmount = IERC20(usn).balanceOf(address(this));
+        require(usnAmount > 0, "StableLpMaker: no Usn to add liquidity");
+
+        IERC20(usn).approve(
+            address(stableSwap),
+            usnAmount
+        );
+
+        uint256[] memory ma = new uint[](3);
+
+        ma[0] = 0;
```

Comment:
what happens if you remove line 120, 121? since an array should be initialized to `0` anyways
just keep line `ma[2] = usnAmount;`

Reply:
works